### PR TITLE
PERF: Cython version of Python _TIMEPAT regexp in parsing.pyx

### PR DESCRIPTION
**Pull request by anmyachev**
- [x] closes #N/A
- [ ] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

`asv continuous -f 1.05 origin/master HEAD -b ^io.csv.ReadCSVParseSpecialDate -a warmup_time=2 -a sample_time=2`:

master | patch | ratio | test_name
-|-|-|-
276±1ms|          262±2ms|     0.95|  io.csv.ReadCSVParseSpecialDate.time_read_special_date('hm')
16.6±0.2ms|      14.2±0.08ms|     0.85|  io.csv.ReadCSVParseSpecialDate.time_read_special_date('mdY')
42.7±0.9ms|       35.0±0.2ms |    0.82|  io.csv.ReadCSVParseSpecialDate.time_read_special_date('mY')

### CODE REVIEW

**Comment by vnlitvin** [comment]
Note to reviewers - current failures seem to be introduced by #26200

File: pandas/_libs/tslibs/parsing.pyx [hunk 1]
```diff
@@ -144,6 +142,26 @@ cdef inline object _parse_delimited_date(object date_string, bint dayfirst):
     raise DateParseError("Invalid date specified ({}/{})".format(month, day))
 
 
+cdef inline bint does_string_look_like_time(object parse_string):
+    cdef:
```
Comment (jreback) [inline pandas/_libs/tslibs/parsing.pyx]: can you add a doc-string here

**Review by jreback — CHANGES_REQUESTED** [review]
otherwise lgtm.

**Comment by jreback** [comment]
also merge master; ci has been fixed

File: pandas/_libs/tslibs/parsing.pyx (hunk 1 above)
Comment (vnlitvin) [reply pandas/_libs/tslibs/parsing.pyx]: docstring added

**Comment by vnlitvin** [comment]
> also merge master; ci has been fixed

I've rebased instead, I personally find rebased history cleaner than merged.

**Comment by jreback** [comment]
> I've rebased instead, I personally find rebased history cleaner than merged.

I agree :-D

we use merge master as its somwhat friendlier to new users, in any event everything is squashed on merge anyways (so the commit history is only relevant to you)

**Comment by vnlitvin** [comment]
@jreback pinging on green

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @anmyachev

**Pull request merged**